inception-search-mtas/src/main/java/de/tudarmstadt/ukp/inception/search/index/mtas/MtasUimaParser.java

```diff
@@ -242,68 +242,158 @@ else if (annotation instanceof Sentence) {
+                                            annotation.getEnd());
+                                    mtasAnnotationFeatureLabel.addPositionRange(beginToken,
+                                            endToken);
+                                    tokenCollection.add(mtasAnnotationFeatureLabel);
+                                    
+                                    // Indexing UI annotation without type and layer for generic search
+                                    indexedStr = IndexingConstants.kbEntity + MtasToken.DELIMITER
+                                            + kbValue;
+                                    log.debug("Indexed String without type and label for : {}", indexedStr);
+                                    MtasToken mtasAnnotationKBEntity = new MtasTokenString(
+                                            mtasId++, indexedStr, beginToken);
+                                    mtasAnnotationKBEntity.setOffset(annotation.getBegin(),
+                                            annotation.getEnd());
+                                    mtasAnnotationKBEntity.addPositionRange(beginToken,
+                                            endToken);
+                                    tokenCollection.add(mtasAnnotationKBEntity);
+                                    
+                                }
+
                             }
                         }
                     }
                 }
             }
-        }
-        catch (Exception e) {
+        } catch (Exception e) {
             log.error("Unable to index document", e);
         }
         return tokenCollection;
     }
-
+    
     @Override
-    public String printConfig()
-    {
+    public String printConfig() {
         return null;
     }
 
-    class SimpleAnnotationLayer
-    {
+    /**
+     * Takes in IRI for identifier and returns teh label String Eg: InputParameter :-
+     * http://www.w3.org/TR/2003/PR-owl-guide-20031209/wine#RoseDAnjou Returned :-
+     * "KBConcept"+MtasToken.DELIMITER+RoseDAnjou
+     * 
+     * @param iri
+     * @return
+     */
+    public String getUILabel(String aIRI) {
+
+        StringBuilder labelStr = new StringBuilder();
+        System.out.println();
+        Optional<KBObject> kbObject = kbUtil.readKBIdentifier(project, aIRI);
+        if (kbObject.isPresent()) {
+            labelStr.append(kbObject.get().getClass().getSimpleName() + MtasToken.DELIMITER
+                    + kbObject.get().getUiLabel());
+        } else {
+            return labelStr.toString();
+        }
+
+        return labelStr.toString();
+    }
+    
+    /**
+     * Method Implementation to get MtasToken (To be done when subclass and class 
+     * semantics to be included
+     * @return
+     */
+    public MtasToken getMtasTokenKBFeature() {
+        return null;
+        
+    }
+
+    class SimpleAnnotationLayer {
         private ArrayList<Pair<String, String>> features;
         private String LayerName;
```

Comment:
Field names start with small letters.

Reply:
As far as I see, `SimpleAnnotationLayer` has not been used. Can this be removed?

Reply:
If it is not used, I guess yes. @betoboullosa , do you know where that SimpleAnnotationLayer comes from?